x-pack/elastic-agent/pkg/agent/application/managed_mode.go

```diff
@@ -147,7 +147,7 @@ func newManaged(
 		router,
 		&pipeline.ConfigModifiers{
 			Decorators: []pipeline.DecoratorFunc{modifiers.InjectMonitoring},
-			Filters:    []pipeline.FilterFunc{filters.StreamChecker, modifiers.InjectFleet(rawConfig, sysInfo.Info(), agentInfo)},
+			Filters:    []pipeline.FilterFunc{filters.StreamChecker, modifiers.InjectInsecureOutput(cfg.Fleet), modifiers.InjectFleet(rawConfig, sysInfo.Info(), agentInfo)},
```

Comment:
This seems like it will affect more than just Fleet Server running under Elastic Agent, it will also affect all the other beats, correct?

If this does affect the other beats, I don't think we want this, because how will this work when it comes to multiple outputs? I believe it will have the effect that if `--insecure` is used all outputs will then become insecure.

Reply:
yes that's the purpose. if we wont pass, events wont get consumed

Reply:
They would get consumed if the policy is configured correctly. I don't think it should be Elastic Agent's job to override the output settings from Kibana.

If we had proper health reporting for outputs, it would be clear there is an issue. The real issue is that the Elastic Agent reports healthy when it is not.

Reply:
if value is provided by kibana it should not be overriden, i'll add testcase to make it more visible

Reply:
kibana on the other hand would needs to have the info that hey this cert we're using is self signed and may no be properly configured. 
insecure is our concept and i think it's up to us to help user using this with FRE.